pkg/agents/team1/agent1/agent.go

```diff
@@ -3,15 +3,21 @@ package agent1
 import (
 	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
 	"github.com/SOMAS2021/SOMAS2021/pkg/messages"
-	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/agent"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/world"
+
+	log "github.com/sirupsen/logrus"
 )
 
 type CustomAgent1 struct {
 	*infra.Base
 	myNumber int
 }
 
-func New(baseAgent *infra.Base) (agent.Agent, error) {
+func New(world world.World, agentType int, agentHP int, agentFloor int, id string) (infra.Agent, error) {
+	baseAgent, err := infra.NewBaseAgent(world, agentType, agentHP, agentFloor, id)
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Didn't want to put this in the custom agent as it makes it easy for custom agents to manipulate their original health etc.